openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java

```diff
@@ -265,6 +267,71 @@ void deleteInjectsForScenarioTest() throws Exception {
 
   // -- EXERCISES --
 
+
+  @DisplayName("Add an inject for simulation")
+  @Test
+  @Order(9)
+  @WithMockPlannerUser
+  void addInjectForSimulationTest() throws Exception {
+    // -- PREPARE --
+    InjectInput input = new InjectInput();
+    input.setTitle("Test inject");
+    input.setInjectorContract(EMAIL_DEFAULT);
+    input.setDependsDuration(0L);
+
+    // -- EXECUTE --
+    String response = mvc
+        .perform(post(EXERCISE_URI + "/" + EXERCISE.getId() + "/injects")
+            .content(asJsonString(input))
+            .contentType(MediaType.APPLICATION_JSON)
+            .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().is2xxSuccessful())
+        .andReturn()
+        .getResponse()
+        .getContentAsString();
+
+    // -- ASSERT --
+    assertNotNull(response);
+    EXERCISE_INJECT_ID = JsonPath.read(response, "$.inject_id");
+    response = mvc
+        .perform(get(EXERCISE_URI + "/" + EXERCISE.getId())
```

Comment:
If this request fails, your test will fail even if the add inject request succeeds. This is unit testing the add inject request test should be independant from the fetch injects request